### PR TITLE
[extension-selenium] Add assertElementsNumber(String,List,ComparisonRule,int) for base validations

### PR DESCRIPTION
**Pull request by uarlouski**

### CODE REVIEW

File: vividus-extension-selenium/src/main/java/org/vividus/bdd/steps/ui/validation/IBaseValidations.java [hunk 1]
```diff
@@ -42,6 +42,15 @@
...
             int number);
 
+    /**
+     * Assert that the number of elements in <b>elements</b> collection is equal to <b>number</b>
+     * @param description description of elements in the <b>elements</b> collection
+     * @param elements collection being verified
+     * @param number expected number of elements in the <b>element</b> collection
+     * @return whether the number of elements in <b>elements</b> is equal to <b>number</b>
+     */
+    boolean assertElementsNumber(String description, List<WebElement> elements, int number);
```
Comment (valfirst) [inline vividus-extension-selenium/src/main/java/org/vividus/bdd/steps/ui/validation/IBaseValidations.java:52]: what about using comparison rule?

File: vividus-extension-selenium/src/main/java/org/vividus/bdd/steps/ui/validation/BaseValidations.java [hunk 2]
```diff
@@ -100,6 +100,13 @@ public boolean assertElementNumber(String businessDescription, String systemDesc
                 elementNumber(Matchers.comparesEqualTo(number)));
     }
 
+    @Override
+    public boolean assertElementsNumber(String description, List<WebElement> elements, int number)
```
Comment (ikalinin1) [inline vividus-extension-selenium/src/main/java/org/vividus/bdd/steps/ui/validation/BaseValidations.java:104]: what's the reason for this method?

Comment (uarlouski) [reply vividus-extension-selenium/src/main/java/org/vividus/bdd/steps/ui/validation/BaseValidations.java:104]: I'll need this in my next PR, additionally it's replacement for a deprecated method

File: vividus-extension-selenium/src/main/java/org/vividus/bdd/steps/ui/validation/BaseValidations.java [hunk 3]
```diff
@@ -100,6 +100,14 @@ public boolean assertElementNumber(String businessDescription, String systemDesc
                 elementNumber(Matchers.comparesEqualTo(number)));
     }
 
+    @Override
+    public boolean assertElementsNumber(String description, List<WebElement> elements, ComparisonRule comparisonRule,
+            int number)
+    {
+        return uiContext.withAssertingWebElements(elements,
+            () -> softAssert.assertThat(description, elements.size(), comparisonRule.getComparisonRule(number)));
+    }
```
Comment (ikalinin1) [inline vividus-extension-selenium/src/main/java/org/vividus/bdd/steps/ui/validation/BaseValidations.java:109]: elements, elementMatcher(comparison rule.getComparisonRule(size))

Comment (uarlouski) [reply vividus-extension-selenium/src/main/java/org/vividus/bdd/steps/ui/validation/BaseValidations.java:109]: Checked it and it turned out not to be properly aligned with comparison rule message, so we get as a result the following:
[Expected: number of elements is a value equal to <...> Actual: <...>]

Comment (uarlouski) [reply vividus-extension-selenium/src/main/java/org/vividus/bdd/steps/ui/validation/BaseValidations.java:109]: [Expected: number of elements is a value less than or equal to <...> Actual: <...>]

**Review by ikalinin1 — APPROVED** [review]

**Pull request merged**